Move list size constants for profiler::Event and profiler::ProfilerConfig into
enum.

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40474 Move list size constants for profiler::Event and profiler::ProfilerConfig into
enum.**
enum.**
enum.**
enum.**
enum.**
enum.**
* #40326 Add callback with TLS state API in futures
enum.**
* #40326 Add callback with TLS state API in futures

These constants are unnecessary since there is an enum, and we can add
the size at the end of the enum and it will be equal to the list size. I
believe that this is the typical pattern used to represent enum sizes.

Differential Revision: [D22147754](https://our.internmc.facebook.com/intern/diff/D22147754/)

rohan-varma committed:
Move list size constants for profiler::Event and profiler::ProfilerConfig into
enum.

These constants are unnecessary since there is an enum, and we can add
the size at the end of the enum and it will be equal to the list size. I
believe that this is the typical pattern used to represent enum sizes.

Differential Revision: [D22147754](https://our.internmc.facebook.com/intern/diff/D22147754/)

[ghstack-poisoned]

ilia-cher commented on torch/csrc/autograd/profiler.cpp:
could you keep the style of enum elements, e.g. ``NUM_EVENT_IVALUE_IDX`` and the same for below

rohan-varma committed:
Update on "Move list size constants for profiler::Event and profiler::ProfilerConfig into
enum."

enum.**
* #40326 Add callback with TLS state API in futures

These constants are unnecessary since there is an enum, and we can add
the size at the end of the enum and it will be equal to the list size. I
believe that this is the typical pattern used to represent enum sizes.

Differential Revision: [D22147754](https://our.internmc.facebook.com/intern/diff/D22147754/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move list size constants for profiler::Event and profiler::ProfilerConfig into
enum."

enum.**
* #40326 Add callback with TLS state API in futures
enum.**
* #40326 Add callback with TLS state API in futures

These constants are unnecessary since there is an enum, and we can add
the size at the end of the enum and it will be equal to the list size. I
believe that this is the typical pattern used to represent enum sizes.

Differential Revision: [D22147754](https://our.internmc.facebook.com/intern/diff/D22147754/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move list size constants for profiler::Event and profiler::ProfilerConfig into
enum."

enum.**
enum.**
* #40326 Add callback with TLS state API in futures
enum.**
* #40326 Add callback with TLS state API in futures

These constants are unnecessary since there is an enum, and we can add
the size at the end of the enum and it will be equal to the list size. I
believe that this is the typical pattern used to represent enum sizes.

Differential Revision: [D22147754](https://our.internmc.facebook.com/intern/diff/D22147754/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move list size constants for profiler::Event and profiler::ProfilerConfig into
enum."

enum.**
enum.**
enum.**
* #40326 Add callback with TLS state API in futures
enum.**
* #40326 Add callback with TLS state API in futures

These constants are unnecessary since there is an enum, and we can add
the size at the end of the enum and it will be equal to the list size. I
believe that this is the typical pattern used to represent enum sizes.

Differential Revision: [D22147754](https://our.internmc.facebook.com/intern/diff/D22147754/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move list size constants for profiler::Event and profiler::ProfilerConfig into
enum."

enum.**
enum.**
enum.**
enum.**
* #40326 Add callback with TLS state API in futures
enum.**
* #40326 Add callback with TLS state API in futures

These constants are unnecessary since there is an enum, and we can add
the size at the end of the enum and it will be equal to the list size. I
believe that this is the typical pattern used to represent enum sizes.

Differential Revision: [D22147754](https://our.internmc.facebook.com/intern/diff/D22147754/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move list size constants for profiler::Event and profiler::ProfilerConfig into
enum."

enum.**
enum.**
enum.**
enum.**
enum.**
* #40326 Add callback with TLS state API in futures
enum.**
* #40326 Add callback with TLS state API in futures

These constants are unnecessary since there is an enum, and we can add
the size at the end of the enum and it will be equal to the list size. I
believe that this is the typical pattern used to represent enum sizes.

Differential Revision: [D22147754](https://our.internmc.facebook.com/intern/diff/D22147754/)

[ghstack-poisoned]
